web/src/settings_panel_menu.js

```diff
@@ -151,17 +159,23 @@ export class SettingsPanelMenu {
         this.$main_elem.children("li").removeClass("active");
         this.$curr_li.addClass("active");
 
-        const settings_section_hash = "#" + this.hash_prefix + section;
+        if (section !== "user-list-admin") {
+            const settings_section_hash = "#" + this.hash_prefix + section;
 
-        // It could be that the hash has already been set.
-        browser_history.update_hash_internally_if_required(settings_section_hash);
+            // It could be that the hash has already been set.
+            browser_history.update_hash_internally_if_required(settings_section_hash);
+        }
 
         $(".settings-section").removeClass("show");
 
         settings_sections.load_settings_section(section);
 
         this.get_panel().addClass("show");
 
+        if (section === "user-list-admin") {
+            right_section = right_section || window.location.hash.split("/")[2];
+            this.get_valid_users_section_panel(right_section);
```

Comment:
Why is this needed? I guess we can assume that caller will make sure that the correct `right_section` variable is passed if `section == "user-list-admin"`.

Reply:
I have written this because initially when the hashchanged url is called , it triggers the `activate_section_or_default` function with the correct `right_section`. But then we call the `[data-user-settings-section].show()` function for rendering the correct section. Now this `show()` function again calls the `activate_section_or_default()` without a `right_section` property. Due to which `right_section` is again handled as `active-users-admin`.

Reply:
I am not sure whether you are talking about jquery `show` or `show()` function defined in `SettingsPanelMenu`. The `show()` function defined in `SettingsPanelMenu` is only called while toggling between personal and organization settings, so I guess we can pass the `right_section` there and it should be fine.

Reply:
Ohh yes that's right @sahil839 .
But then I could not find the reason why the `activate_section_or_default` function is called two times when I use hashchange to trigger it.
Due to this the first time , the call is made as expected but for the second time the right_section is found to be undefined which ultimately shows the `active-users-admin` section.